resources/js/contest-voting/base-entry-list.coffee

```diff
@@ -44,10 +51,19 @@ export class BaseEntryList extends React.Component
       waitingForResponse: false
       callback
 
+
   componentDidMount: ->
     $.subscribe "contest:vote:click.#{@eventId}", @handleVoteClick
     $.subscribe "contest:vote:done.#{@eventId}", @handleUpdate
 
+
+  componentDidUpdate: (_, prevState) ->
+    if !isEqual(prevState.selected, @state.selected) || prevState.showVotedOnly != @state.showVotedOnly
```

Comment:
looking again, the previous one is probably more correct in term of dealing with `waitingForResponse` state which affects the `selected` array 👀 

Or maybe add the state check (which prevents saving incorrect `selected` state on toggling `showVotedOnly` while also waiting for response)

Or don't mix selected state with inflight one (probably requires larger change)... And I just noticed there's nothing handling failed voting request?

Reply:
I forgot that's why I put it in the callback in the first place 👀